### ISSUE TITLE
Display issue with HTML Block inside of a Row Block in the Editor

### ISSUE THREAD

**Issue by ndiego** [WordPress/gutenberg#34909]
### Description

I was testing the new Row Block and discovered a small issue in the editor when using an HTML block. The HTML content is contained in an iframe and also seems to be wrapping in a `div`. This iframe and `div` obviously do not appear on the frontend. In the editor however, they both have a width of 100%. Under normal circumstances, this seems like a logical default. However, when placed in a Row block, things do not align properly. 

### Step-by-step reproduction instructions

1. Create a Row block.
2. Add a HTML block and add some content, I added an SVG icon.
3. Make sure the HTML block is in "Preview mode".
3. Add another block to the Row block, I chose a simple Heading block.
4. Notice the ample space between the content in the HTML block and the Heading. 
5. Switch to the frontend, and notice this spacing is not there. 

**Editor (Actual)**

![image](https://user-images.githubusercontent.com/4832319/133785302-c4ee60fc-ed6b-4641-92bd-3c4e41cf059e.png)

**Editor (Expected)**

![image](https://user-images.githubusercontent.com/4832319/133787165-f2200d00-d060-4257-ba43-2dc4fa0dd1d6.png)

**Frontend**

![image](https://user-images.githubusercontent.com/4832319/133785358-90a6e857-51e0-44d2-a66f-655d991f6460.png)

**Some of the offending CSS**

![image](https://user-images.githubusercontent.com/4832319/133785674-7471b8d7-3f0a-4307-9383-e5528f993d0c.png)

Perhaps there is a need for some conditional iframe styling if placed in a Row block? Perhaps `max-width:100%` instead of `width:100%`?

### Environment info

- WordPress 5.8.1
- Gutenberg 11.5
- Chrome/Safari

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alaczek** [comment]
I'm experiencing a similar issue with Social Icons and Paragraph block nested inside the Row block.

Editor view:
<img width="675" alt="image" src="https://user-images.githubusercontent.com/4550351/185283250-08623083-156b-4212-bf24-3abd67d724c6.png">

Front end:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/4550351/185283322-4dbef0c9-54e5-4e55-bdd4-c4d308cbee8a.png">

In my case, the offending piece of CSS seems to be related to the fact that the Row block is inside a Cover block, if I place the Row block on it's own, the issue goes away. As far as I can tell, this is the piece of CSS that's causing it:
<img width="371" alt="image" src="https://user-images.githubusercontent.com/4550351/185283977-babff85f-0e39-4d62-8fb1-bf0960c3f1f6.png">


Code snippet:
```
<!-- wp:cover {"customOverlayColor":"#7749f9","minHeight":100,"minHeightUnit":"vh","align":"full","style":{"spacing":{"padding":{"top":"8vh","right":"5vh","bottom":"8vh","left":"5vh"}}}} -->
<div class="wp-block-cover alignfull" style="padding-top:8vh;padding-right:5vh;padding-bottom:8vh;padding-left:5vh;min-height:100vh"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-100 has-background-dim" style="background-color:#7749f9"></span><div class="wp-block-cover__inner-container"><!-- wp:group {"style":{"spacing":{"blockGap":"64px"},"border":{"radius":"100px"}},"backgroundColor":"secondary","layout":{"inherit":true}} -->
<div class="wp-block-group has-secondary-background-color has-background" style="border-radius:100px"><!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left"}} -->
<div class="wp-block-group"><!-- wp:social-links {"className":"is-style-default"} -->
<ul class="wp-block-social-links is-style-default"><!-- wp:social-link {"url":"#","service":"instagram"} /--></ul>
<!-- /wp:social-links -->

<!-- wp:paragraph -->
<p><a href="#">Instagram</a></p>
<!-- /wp:paragraph --></div>
<!-- /wp:group --></div>
<!-- /wp:group --></div></div>
<!-- /wp:cover -->
```

**Environment info**
* WordPress 6.0.1
* Gutenberg 13.9
* Chrome 104.0

**Comment by jasmussen** [comment]
@alaczek I tried your fix in #43413. I'd appreciate a ton of testing of various cover patterns.
